autotest/tools/restful/test_restful_chat_workspace.py

```diff
@@ -0,0 +1,197 @@
+import os
+import subprocess
+from time import sleep, time
+
+import allure
+import pytest
+from pytest import assume
+from utils.config_utils import (get_cuda_prefix_by_workerid,
+                                get_turbomind_model_list, get_workerid)
+from utils.get_run_config import get_command_with_extra
+from utils.run_client_chat import command_line_test
+from utils.run_restful_chat import (get_model, health_check, interactive_test,
+                                    open_chat_test)
+
+BASE_HTTP_URL = 'http://localhost'
+DEFAULT_PORT = 23333
+
+
+@pytest.fixture(scope='function', autouse=True)
+def prepare_environment(request, config, worker_id):
+    dst_path = config.get('dst_path')
+    log_path = config.get('log_path')
+
+    param = request.param
+    model = param['model']
+    cuda_prefix = param['cuda_prefix']
+    tp_num = param['tp_num']
+
+    if cuda_prefix is None:
+        cuda_prefix = get_cuda_prefix_by_workerid(worker_id, tp_num=tp_num)
+
+    worker_num = get_workerid(worker_id)
+    if worker_num is None:
+        port = DEFAULT_PORT
+    else:
+        port = DEFAULT_PORT + worker_num
+
+    cmd = get_command_with_extra('lmdeploy serve api_server ' + dst_path +
+                                 '/workspace_' + model + ' --server-port ' +
+                                 str(port),
+                                 config,
+                                 model,
+                                 need_tp=True,
+                                 cuda_prefix=cuda_prefix)
+
+    if 'kvint8' in model:
+        cmd += ' --quant-policy 4'
+        if 'w4' in model or '4bits' in model:
+            cmd += ' --model-format awq'
+        else:
+            cmd += ' --model-format hf'
+    if 'w4' in model or '4bits' in model:
```

Comment:
duplicate `--model-format awq` if both `kvint8` and `w4` or `4bits` are in the `model`